examples/multimodal/multimodal_llm/neva/neva_evaluation.py

```diff
@@ -0,0 +1,358 @@
+
+    c. run top_p inference from a nemo file:
+        python neva_evaluation.py \
+            neva_model_file=PATH_TO_MODEL \
+            inference.greedy=False \
+            inference.top_k=0 \
+            inference.top_p=0.9 \
+            inference.repetition_penalty=1.2 \
+            inference.add_BOS=True \
+            trainer.devices=1 \
+            trainer.num_nodes=1 \
+            tensor_model_parallel_size=-1 \
+            pipeline_model_parallel_size=-1 \
+            prompts=[prompt1,prompt2]
+
+    d. If you don't need to generate tokens and need model to compute logprobs:
+         python neva_evaluation.py \
+            neva_model_file=PATH_TO_MODEL \
+            inference.compute_logprob=True \
+            trainer.devices=1 \
+            trainer.num_nodes=1 \
+            tensor_model_parallel_size=-1 \
+            pipeline_model_parallel_size=-1 \
+            prompts=[text to get logprob]
+
+    e. Launch the inference server
+         python neva_evaluation.py \
+            neva_model_file=PATH_TO_MODEL \
+            trainer.devices=1 \
+            trainer.num_nodes=1 \
+            tensor_model_parallel_size=-1 \
+            pipeline_model_parallel_size=-1 \
+            server=True
+        
+        To send a request to the server, here is one example code:
+        ```python
+        import json
+        import requests
+
+        batch_size = 8
+        port_num = 5555
+        headers = {"Content-Type": "application/json"}
+
+
+        def request_data(data):
+            resp = requests.put('http://localhost:{}/generate'.format(port_num),
+                                data=json.dumps(data),
+                                headers=headers)
+            sentences = resp.json()['sentences']
+            return sentences
+
+
+        data = {
+            "sentences": [""] * batch_size,
+            "images" : [] * batch_size,
+            "tokens_to_generate": 300,
+            "temperature": 1.0,
+            "add_BOS": True,
+            "top_k": 0,
+            "top_p": 0.9,
+            "greedy": False,
+            "all_probs": False,
+            "repetition_penalty": 1.2,
+            "min_tokens_to_generate": 2,
+        }
+
+        sentences = request_data(data)
+        ```
+"""
+
+if not torch.cuda.is_available():
+    raise EnvironmentError("GPU is needed for the inference")
+
+
+class RequestDataSet(Dataset):
+    def __init__(self, sentences):
+        super().__init__()
+        self.sentences = sentences
+
+    def __len__(self,):
+        return len(self.sentences)
+
+    def __getitem__(self, idx):
+        return self.sentences[idx]
+
+
+@hydra_runner(config_path="conf", config_name="neva_inference")
+def main(cfg) -> None:
+
+    plugins = []
+    if cfg.get('cluster_type', None) == 'BCP':
+        plugins.append(TorchElasticEnvironment())
+    # trainer required for restoring model parallel models
+    trainer = Trainer(plugins=plugins, strategy=NLPDDPStrategy(), **cfg.trainer)
```

Comment:
Can be in a later PR but we should use the TrainerBuilder for all example scripts

Reply:
Yeah, I refactored this part in current code. Will come in later PR.